File: frontend/src/types/gpu.ts
// GPU メトリクスのデータ型定義
export interface GPUMetrics {
  node_name: string;
  gpu_index: number;
  gpu_name: string;
  utilization: number;
  memory_used: number;
  memory_total: number;
  memory_utilization: number;
  temperature: number;
  power_draw: number;
  power_limit: number;
  timestamp: string;
}

// GPU ノード情報の型定義
export interface GPUNode {
  node_name: string;
  gpu_count: number;
  gpu_models: string[];
}

// API レスポンスの型定義
export interface APIResponse<T> {
  success: boolean;
  data?: T;
  message?: string;
  error?: string;
}

// GPU 利用率の型定義
export interface GPUUtilization {
  node: string;
  gpu_index: string;
  utilization: string;
  timestamp: number;
}

// テーブルの設定タイプ
export interface TableConfig {
  pageSize: number;
  refreshInterval: number;
  autoRefresh: boolean;
}

// フィルタリングの設定
export interface FilterConfig {
  nodeFilter: string;
  utilizationRange: [number, number];
  temperatureRange: [number, number];
  showOnlyHighUtilization: boolean;
} 

File: backend/internal/models/gpu.go
package models

import "time"

// GPUMetrics represents GPU metrics data structure
type GPUMetrics struct {
	NodeName          string    `json:"node_name"`
	GPUIndex          int       `json:"gpu_index"`
	GPUName           string    `json:"gpu_name"`
	Utilization       float64   `json:"utilization"`
	MemoryUsed        float64   `json:"memory_used"`
	MemoryTotal       float64   `json:"memory_total"`
	MemoryFree        float64   `json:"memory_free"`
	MemoryUtilization float64   `json:"memory_utilization"`
	Temperature       float64   `json:"temperature"`
	PowerDraw         float64   `json:"power_draw"`
	PowerLimit        float64   `json:"power_limit"`
	Timestamp         time.Time `json:"timestamp"`
}

// GPUNode represents GPU node information
type GPUNode struct {
	NodeName  string   `json:"node_name"`
	GPUCount  int      `json:"gpu_count"`
	GPUModels []string `json:"gpu_models"`
}

// APIResponse represents standard API response structure
type APIResponse struct {
	Success bool        `json:"success"`
	Data    interface{} `json:"data,omitempty"`
	Error   string      `json:"error,omitempty"`
	Message string      `json:"message,omitempty"`
}

// MetricsQuery represents Prometheus query parameters
type MetricsQuery struct {
	Query     string `json:"query"`
	StartTime string `json:"start_time,omitempty"`
	EndTime   string `json:"end_time,omitempty"`
	Step      string `json:"step,omitempty"`
}
